app/models/repository/validations.rb

```diff
@@ -20,76 +18,111 @@ module Repository::Validations
                      if: :path_changed?
 
     validates_with UnreservedValidator, if: :path_changed?
+
+    validates_inclusion_of :state, in: Repository::Importing::STATES
+    validates_with SourceTypeValidator, if: :source_address?
+    validates_presence_of :remote_type, if: :source_address?
+    validates_with RemoteTypeValidator
+
+    validates_with RemoteAccessValidator
+    validates :access,
+      presence: true,
+      inclusion: { in: Repository::Access::OPTIONS }
   end
 
-  def set_path
-    self.path ||= name.parameterize if name
+  class RemoteAccessValidator < ActiveModel::Validator
+    def validate(record)
+      if record.mirror? && (record.private_rw? || record.public_rw?)
+        record.errors[:access] = "Error! Write access is not allowed for a mirrored repositry."
```

Comment:
Metrics/LineLength: Line is too long. [95/80]<br>Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.